polaris-service/src/main/java/org/apache/polaris/service/persistence/cache/EntityCacheFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.persistence.cache;
+
+import jakarta.inject.Inject;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.context.RealmScope;
+import org.apache.polaris.core.persistence.PolarisMetaStoreManager;
+import org.apache.polaris.core.persistence.cache.EntityCache;
+import org.glassfish.hk2.api.Factory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityCacheFactory implements Factory<EntityCache> {
```

Comment:
Suggestion: `jakarta.enterprise.inject.Produces` from jakarta.enterprise.cdi-api 4.1.0

Reply:
HK2 doesn't implement support for `@Produces`. It does ship with its own `@Provides` annotation (see https://github.com/eclipse-ee4j/glassfish-hk2/pull/488 ), but that annotation is HK2-specific and doesn't really solve the intended goal, I think. It's possible to add something, but I think we should avoid cluttering this PR and leave that as an improvement to make afterward.

Reply:
Well, in this case it looks like factories still depend on HK2 classes heavily, which will complicate reuse (HK2 will trickle down all downstream runtime environments). I'm not sure what the benefit of using HK2 will be if we cannot make it a runtime-only dependency, but it effectively becomes an API dependency (meaning any attempt to customize DI will have to bind to HK2 APIs).

Reply:
The HK2 factory classes can be restricted to the dropwizard runtime. There are no references to these factory classes outside of the `PolarisApplication`, which is Dropwizard-specific